### DISCUSSION TITLE
brew link python@3.9 doesn't create /usr/local/bin/{python3,pip3,...}

### DISCUSSION THREAD

**Discussion by ed-randall-blk** [Homebrew/discussions#4785] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.11-17-g24de357
ORIGIN: https://github.com/Homebrew/brew
HEAD: 24de357fbf5901d950cfa73acbffcbb93b4aba7f
Last commit: 2 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: ab676d59d4bc5a4207c61341735b415bb7d54a5d
Core tap last commit: 71 minutes ago
Core tap branch: master
Core tap JSON: 13 Sep 09:28 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 16
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_INSTALL_CLEANUP: set
HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 2.6.10 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.5.2-x86_64
CLT: 14.3.1.0.1.1683849156
Xcode: 14.3
```

### Output of `brew doctor`

```shell
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  go@1.19
  python@3.7

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  go
  tcl-tk
  python@3.10
  python@3.8
  python@3.7
  python@3.9
```

### Description of issue

`brew link python@3.9` doesn't create the following expected links from `/usr/local/bin/{file}` to `Cellar/python@3.9/3.9.18/bin/{file}` for the following {file}s:

- 2to3
- idle3
- pip3
- pydoc3
- python3
- wheel3
- pkgconfig/python3-embed.pc
- pkgconfig/python3.pc

This is inconsistent with python@3.7, @3.8, @3.10, all of which create these links.
Users have to be aware and take steps to run `/usr/local/bin/python3.9` rather than `/usr/local/bin/python3` which rather defeats the utility of having the links.

**Comment by ZhongRuoyu** [comment]
It shouldn't be (unless your 3.7, 3.8, and 3.10 installations are outdated), because only `python@3.11` provides the those links that you listed.

What's the output of the following:

```shell
ls "$(brew --prefix)"/opt/python@3.{7..10}/{bin,lib/pkgconfig}
brew outdated
```

**Reply by edrandall** [reply]
But... that's... not... useful.   Why the behaviour change?
We're not on python 3.11... we're only just upgrading from 3.8 to 3.9 ;)

Obviously occasionally we have to upgrade, but generally we favour stability over having the latest of everything all the time - hence the HOMEBREW_NO_AUTO_UPDATE and other settings.

**Reply by SMillerDev** [reply]
If you prefer stability, pyenv or docker is probably a better fit than trying to treat Homebrews ever changing software landscape as immutable.

**Reply by edrandall** [reply]
OK, pyenv seems to do what I need.   But... now have multiple installs of similar versions of Python... why ... it's possible to symlink in different versions of `go` and `node` just fine using `brew link` without a `goenv` or `nodeenv` (for example).   Seems rather inconsistent.
